runtimes/core/src/api/gateway/mod.rs

```diff
@@ -192,20 +196,37 @@ impl ProxyHttp for Gateway {
     ) -> pingora::Result<Box<HttpPeer>> {
         let path = session.req_header().uri.path();
 
+        // Check if this is a pubsub push request and if we need to proxy it to another service
+        let push_proxy_svc = path
+            .strip_prefix("/__encore/pubsub/push/")
+            .and_then(|sub_id| self.inner.proxied_push_subs.get(sub_id));
+
         if let Some(own_api_addr) = &self.inner.own_api_address {
-            if path.starts_with("/__encore/") {
+            if push_proxy_svc.is_none() && path.starts_with("/__encore/") {
                 return Ok(Box::new(HttpPeer::new(own_api_addr, false, "".to_string())));
             }
         }
-
-        let method: Method = session
-            .req_header()
-            .method
-            .as_ref()
-            .try_into()
-            .or_err(ErrorType::HTTPStatus(400), "invalid http method")?;
-
-        let service_name = self.inner.router.route_to_service(method, path)?;
+        let service_name = push_proxy_svc
+            .map(Ok)
+            .or_else(|| {
```

Comment:
I think this can be `map_or_else(|| { ... }, Ok)` instead. Or a match statement, which might be clearer?